src/Agent.Plugins/TfsVCCliManager.cs

```diff
@@ -244,10 +284,32 @@ protected async Task<TfsVCPorcelainCommandResult> TryRunPorcelainCommandAsync(Fo
                     result.Exception = ex;
                 }
 
+                if(useSecretParameterPassing)
+                {
+                    CleanupTfsVCOutput(ref result, formattedArguments);
+                    File.Delete(Path.Combine(this.SourcesDirectory, cmdFileName));
+                }
+
                 return result;
             }
         }
 
+        private void CleanupTfsVCOutput(ref TfsVCPorcelainCommandResult command, string executedCommand)
+        {
+            List<string> stringsToRemove = new List<string>();
+            foreach(var item in command.Output)
+            {
+                if(item.Contains(executedCommand))
+                {
+                    stringsToRemove.Add(item);
+                }
+            }
+            foreach(var item in stringsToRemove)
+            {
+                command.Output.Remove(item);
+            }
```

Comment:
It can be shortened: 
var stringsToRemove = command.Output.Where(item => item.Contains(executedCommand));

command.Output.RemoveAll(item => stringsToRemove.Contains(item));